### PR TITLE
is_tarfile accepts pathlike

### PR DESCRIPTION
**Pull request by isidentical**

### CODE REVIEW

File: stdlib/2and3/tarfile.pyi [hunk 1]
```diff
@@ -135,7 +135,7 @@ class TarFile(Iterable[TarInfo]):
                    fileobj: Optional[IO[bytes]] = ...) -> TarInfo: ...
     def close(self) -> None: ...
 
-def is_tarfile(name: str) -> bool: ...
+def is_tarfile(name: Union[_Path, IO[bytes]]) -> bool: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/tarfile.pyi]: Are you sure about the `IO[bytes]` part? I got a TypeError when I tried it.

**Comment by isidentical** [comment]
It takes filelike objects as described here (directly passes to
tarfile.open).
https://github.com/python/cpython/blob/master/Lib/tarfile.py#L2471

**Comment by hauntsaninja** [comment]
That's only true of master / 3.9, https://bugs.python.org/issue29435

**Comment by isidentical** [comment]
Thanks, @hauntsaninja.

**Pull request merged**